tutorials/physics/rigid_body.rst

```diff
@@ -29,49 +29,53 @@ The "look at" method
 
  .. code-tab:: csharp
 
     using Godot;
+    using System;
 
     public partial class MyRigidBody3D : RigidBody3D
     {
-        private void LookFollow(PhysicsDirectBodyState state, Transform3D currentTransform, Vector3 targetPosition)
+        float speed = 0.1F;
+        private void LookFollow(PhysicsDirectBodyState3D state, Transform3D currentTransform, Vector3 targetPosition)
         {
-            var upDir = new Vector3(0, 1, 0);
-            var curDir = currentTransform.Basis * new Vector3(0, 0, 1);
-            var targetDir = currentTransform.Origin.DirectionTo(targetPosition);
-            var rotationAngle = Mathf.Acos(curDir.X) - Mathf.Acos(targetDir.X);
-
-            state.SetAngularVelocity(upDir * (rotationAngle / state.GetStep()));
+            Vector3 forwardLocalAxis = new Vector3(1, 0, 0);
+            Vector3 forwardDir = (currentTransform.Basis * forwardLocalAxis).Normalized();
+            Vector3 targetDir = (targetPosition - currentTransform.Origin).Normalized();
+            float local_speed = (float)Math.Clamp(speed, 0.0, Math.Acos(forwardDir.Dot(targetDir)));
+            if (forwardDir.Dot(targetDir) > 1e-4)
+            {
+                state.AngularVelocity = forwardDir.Cross(targetDir) * local_speed / state.Step;
+            }
         }
 
-        public override void _IntegrateForces(PhysicsDirectBodyState state)
+        public override void _IntegrateForces(PhysicsDirectBodyState3D state)
         {
-            var targetPosition = GetNode<Node3D>("MyTargetNode3DNode").GetGlobalTransform().Origin;
-            LookFollow(state, GetGlobalTransform(), targetPosition);
+            Vector3 targetPosition = GetNode<Node3D>("MyTargetNode3DNode").GlobalTransform.Origin;
+            LookFollow(state, GlobalTransform, targetPosition);
         }
     }
 
 
-This method uses the rigid body's ``angular_velocity`` property to rotate the body. It first calculates the difference between the current and desired angle and then adds the velocity needed to rotate by that amount in one frame's time.
-
-.. note:: This script will not work with rigid bodies in *character mode* because then, the body's rotation is locked. In that case, you would have to rotate the attached mesh node instead using the standard Node3D methods.
+This method uses the rigid body's ``angular_velocity`` property to rotate the body. The axis to rotate around is given by the cross product between the current forward direction and the direction one wants to look in. The ``clamp`` is a simple method used to prevent the amount of rotation from going past the direction which is wanted to be looked in, as the total amount of rotation needed is given by the arccosine of the dot product. This method can be used with ``axis_lock_angular_*`` as well. If more precise control is needed, solutions such as ones relying on :ref:`class_Quaternion` may be required, as discussed in :ref:`doc_using_transforms`.
```

Comment:
This last line needs to be broken up, we can't have extremely long lines like this in the file, even if it looks fine on the website. In addition there needs to be a newline at the end of this file, the CI build issue is because of that.

Reply:
Thank you for pointing this out, it has been broken up